src/coreclr/jit/optimizer.cpp

```diff
@@ -4598,6 +4598,352 @@ PhaseStatus Compiler::optUnrollLoops()
+        {
+            fgValueNumberSsaVarDef(falseInput->AsLclVarCommon());
+        }
+    }
+
+    // Invert the condition.
+    cond->gtOper = GenTree::ReverseRelop(cond->gtOper);
+
+    // Create a select node.
+    GenTreeConditional* select =
+        gtNewConditionalNode(GT_SELECT, cond, asgNode->gtGetOp2(), falseInput, asgNode->TypeGet());
+
+    // Use the select as the source of the assignment.
+    asgNode->AsOp()->gtOp2 = select;
+    asgNode->AsOp()->gtFlags |= (select->gtFlags & GTF_ALL_EFFECT);
+    gtSetEvalOrder(asgNode);
+    fgSetStmtSeq(asgStmt);
+
+    // Remove the JTRUE statement.
+    last->ReplaceWith(gtNewNothingNode(), this);
+    gtSetEvalOrder(last);
+    fgSetStmtSeq(block->lastStmt());
+
+    // Before moving anything, fix up any SSAs in the asgBlock
+    for (Statement* const stmt : asgBlock->Statements())
+    {
+        for (GenTree* const node : stmt->TreeList())
+        {
+            if (node->IsLocal())
+            {
+                GenTreeLclVarCommon* lclVar = node->AsLclVarCommon();
+                unsigned             lclNum = lclVar->GetLclNum();
+                unsigned             ssaNum = lclVar->GetSsaNum();
+                if (ssaNum != SsaConfig::RESERVED_SSA_NUM)
+                {
+                    LclSsaVarDsc* ssaDef = lvaGetDesc(lclNum)->GetPerSsaData(ssaNum);
+                    if (ssaDef->GetBlock() == asgBlock)
+                    {
+                        JITDUMP("SSA def %d for V%02u moved from " FMT_BB " to " FMT_BB ".\n", ssaNum, lclNum,
+                                ssaDef->GetBlock()->bbNum, block->bbNum);
+                        ssaDef->SetBlock(block);
+                    }
+                }
+            }
+        }
+    }
+
+    // Move the Asg to the end of the original block
+    Statement* stmtList1 = block->firstStmt();
+    Statement* stmtList2 = asgBlock->firstStmt();
+    Statement* stmtLast1 = block->lastStmt();
+    Statement* stmtLast2 = asgBlock->lastStmt();
+    stmtLast1->SetNextStmt(stmtList2);
+    stmtList2->SetPrevStmt(stmtLast1);
+    stmtList1->SetPrevStmt(stmtLast2);
+    asgBlock->bbStmtList = nullptr;
+
+    // Update the flow from the original block.
+    fgRemoveAllRefPreds(block->bbNext, block);
+    block->bbJumpKind = BBJ_ALWAYS;
+
+#ifdef DEBUG
+    if (verbose)
+    {
+        JITDUMP("\nAfter if conversion\n");
+        fgDumpBlock(block);
+        for (BasicBlock* dumpBlock = block->bbNext; dumpBlock != finalBlock; dumpBlock = dumpBlock->GetUniqueSucc())
+        {
+            fgDumpBlock(dumpBlock);
+        }
+        JITDUMP("\n");
+    }
+#endif
+
+    return true;
+#endif
+}
+
+//-----------------------------------------------------------------------------
+// optIfConversion: If conversion
+//
+// Returns:
+//   suitable phase status
+//
+PhaseStatus Compiler::optIfConversion()
+{
+    bool madeChanges = false;
+
+    // Reverse iterate through the blocks.
+    BasicBlock* block = fgLastBB;
+    while (block != nullptr)
```

Comment:
Not for this PR, but just a thought - Is there a quick way to check if the `block` doesn't definitely have conditional in the end that can be optimized? I am trying to see if have information that we can do a quick check here to eliminate calls to `optIfConvert()` and doing analysis only to find out that block cannot be optimized with if-convert.